### ISSUE TITLE
Large Difference in Results between P100 and V100

### ISSUE THREAD

**Issue by diviramon** [YeongHyeon/GANomaly-PyTorch#1]
Hi guys,

I have been using your network for a bit and things were working really well when I used a Tesla GPU P100. My results were around 0.90 auc.

However, when I switched to a Tesla V100 the best auc I can get is 0.75. The code is identically the same, the only difference is the gpu. Any ideas of why this could happen? 

Here is the env info:

    'gcc_version',
    'clang_version',
    'cmake_version',
    'os',
    'python_version',
    'is_cuda_available',
    'cuda_runtime_version',
    'nvidia_driver_version',
    'nvidia_gpu_models',
    'cudnn_version',

Collecting environment information...
PyTorch version: 1.7.0+cu101
Is debug build: True
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A
OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.12.0
Python version: 3.6 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: Tesla V100-SXM2-16GB
Nvidia driver version: 460.32.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytorch-msssim==0.2.1
[pip3] torch==1.7.0+cu101
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.3.1
[pip3] torchvision==0.8.1+cu101
[conda] Could not collect

It is the same on the P100 with the exception of the GPU. Thanks!

**Comment by YeongHyeon** [comment]
Hi, David Ramon Prados,

In my opinion, it seems that the difference in the backpropagation result caused by the difference in computational precision due to the GPU architecture. For example the following sentence mentions the difference of two GPU, P100 and V100, via FP16 or FP32.

'On P100, half-precision (FP16) FLOPs are reported. On V100, tensor FLOPs are reported, which run on the Tensor Cores in mixed precision: a matrix multiplication in FP16 and accumulation in FP32 precision.'
(refer: https://www.xcelerit.com/computing-benchmarks/insights/benchmarks-deep-learning-nvidia-p100-vs-v100-gpu/)

Thank you for your feedback.
Have a nice day!
:)

**Comment by diviramon** [comment]
Actually I solved the issue by adding:

`torch.set_deterministic(True)`

Thanks a lot for your help!

**Comment by diviramon** [comment]
Hi,

Sorry for a quick follow up. But I was wondering when during test,  `self.netg.eval()` is never called. This would lead to data leakage since the batchnorms would adjust statistics to the testing dataset.

Thanks!

**Comment by YeongHyeon** [comment]
Glad to see you again.

In my repository, the source code does not use the variable 'netg'.
The generator, in this source, named as decoder.
Line 16 @ https://github.com/YeongHyeon/GANomaly-PyTorch/blob/master/source/neuralnet.py

By the way, how about star or fork this repository?

Thank you!

**Comment by diviramon** [comment]
Thanks!
